### ISSUE TITLE
Framework Assemblies missing from package despite being referenced in project file

### ISSUE THREAD

**Issue by bording** [dotnet/sdk#1469]
While #1379 fixed the problem of framework assemblies of dependencies being incorrectly hoisted into the parent nuspec, it appears the there is still a problem in this area.

#### The problem scenario

I have a package that includes source files via the contentFiles mechanism, and it also has framework assembly references to ensure the included source compiles:
```
<frameworkAssemblies>
  <frameworkAssembly assemblyName="System.Configuration" targetFramework=".NETFramework4.5.2" />
  <frameworkAssembly assemblyName="System.Security" targetFramework=".NETFramework4.5.2" />
</frameworkAssemblies>
```
I've added a PackageReference to this package in my project:
```
<PackageReference Include="Particular.Licensing.Sources" Version="1.0.0-alpha0002" PrivateAssets="All" />
```
It's been marked PrivateAssets="All" because it includes source files that I compile into the project, so I don't want it to be a dependency of my project's package.

However, to ensure that all of the framework assembly references are included as part of my project's package, I've explicitly added references as well, along with a third reference required by the project:
```
<ItemGroup Condition="'$(TargetFramework)' == 'net452'">
  <Reference Include="System.Configuration" />
  <Reference Include="System.Security" />
  <Reference Include="System.Transactions" />
</ItemGroup>
```
When I pack this project, I only get a frameworkAssembly entry for System.Transactions. The other two are now missing after upgrading to the latest SDK daily:
```
<frameworkAssemblies>
  <frameworkAssembly assemblyName="System.Transactions" targetFramework=".NETFramework4.5.2" />
</frameworkAssemblies>
```

#### Expected Behavior
I expect to see all of the added references in my project show up as framework references regardless of what packages I have installed, especially if the conflicting package is marked PrivateAssets="All".

#### Actual Behavior
The references are missing if a dependency has them listed as frameworkAssemblies regardless of the PrivateAssets value.

Tested with SDK 2.0.0-preview3-006912

**Comment by bording** [comment]
@rohit21agrawal I filed this against the SDK, but could this be something in the Pack tasks from NuGet instead?

**Comment by rohit21agrawal** [comment]
@bording was this working in an earlier version of sdk?

**Comment by bording** [comment]
@rohit21agrawal It's hard to tell. We went from a bug where frameworkassemblies weren't being included in the package at all, to bugs that caused too many to show up (both implicit references and the references from dependencies were being included). to the current state.

**Comment by bording** [comment]
Digging into this a bit, I believe the problem is actually in the ResolveAssemblyReferences target. Both the direct reference from the project file and the one from the NuGet package get sent in to the target as input, and it appears that the NuGet package copy is the one that is chosen. The ReferencePath output item has all of the metadata of the NuGet copy.

That explains why the change made in #1379 made them disappear, since it now adds Pack=false to the metadata.

Would there be a way to work around this behavior at the SDK level, or does this mean this would have to be fixed/changed directly in the ResolveAssemblyReferences target?

**Comment by bording** [comment]
Ah, and now that I've done that research, it looks like this has the same underlying cause as #1443.

**Comment by nguerrera** [comment]
Tracking the work to fix this in #1443 which indeed has the same root cause. I made a note to capture both issues in tests when fixing.
